src/librustc_trans/trans/consts.rs

```diff
@@ -591,6 +591,21 @@ fn const_expr_unadjusted<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
             let is_float = ty.is_fp();
             let signed = ty.is_signed();
 
+            if ty.is_bool() && (b.node == hir::BiOr || b.node == hir::BiAnd) {
+              match eval_const_expr_partial(cx.tcx(),
```

Comment:
It's really hard to extract the proper value from a llvm constant value. There's stackoverflow posts on it for doing it with c++

http://stackoverflow.com/questions/5315176/llvm-get-constant-integer-back-from-value
http://stackoverflow.com/questions/17255699/outputting-inputs-constant-char-array-from-llvm-pass
http://stackoverflow.com/questions/30314417/llvm-retrieving-the-value-of-constant-function-parameters-from-the-c-api